"[expression.unparsable.type.invalid] Expression invalid in dependent type annotation" in unannotated code

Steps to reproduce:

```
git clone --depth 1 --branch checker-curiosity https://github.com/kennknowles/beam
cd beam
./gradlew compileJava
```

Output:

```
/Users/klk/GitHub/apache/beam/runners/core-java/src/main/java/org/apache/beam/runners/core/InMemoryStateInternals.java:504: error: [expression.unparsable.type.invalid] Expression invalid in dependent type annotation: [error for expression: this.contents.subMap(minTimestamp, true, limitTimestamp, false); error: Invalid 'true' because is not a supported expression]
          .flatMap(e -> e.getValue().stream().map(v -> TimestampedValue.of(v, e.getKey())))
                   ^
```

Since this is unannotated code, I do not really know how to interpret this error message.

--- a/checker/tests/nullness/Issue3792.java
+++ b/checker/tests/nullness/Issue3792.java
@@ -1,0 +1,16 @@
+import java.util.Collection;
+import java.util.NavigableMap;
+
+public abstract class Issue3792<T> {
+    static class Instant {}
+
+    void method(
+            NavigableMap<Instant, Collection<T>> contents,
+            Instant minTimestamp,
+            Instant limitTimestamp) {
+        contents.subMap(minTimestamp, true, limitTimestamp, false).entrySet().stream()
+                .flatMap(e -> e.getValue().stream().map(v -> of(v, e.getKey())));
+    }
+
+    abstract Object of(T v, Instant key);
+}

--- a/framework/src/main/java/org/checkerframework/framework/util/FlowExpressionParseUtil.java
+++ b/framework/src/main/java/org/checkerframework/framework/util/FlowExpressionParseUtil.java
@@ -5,6 +5,7 @@ import com.github.javaparser.StaticJavaParser;
 import com.github.javaparser.ast.ArrayCreationLevel;
 import com.github.javaparser.ast.expr.ArrayAccessExpr;
 import com.github.javaparser.ast.expr.ArrayCreationExpr;
+import com.github.javaparser.ast.expr.BooleanLiteralExpr;
 import com.github.javaparser.ast.expr.CharLiteralExpr;
 import com.github.javaparser.ast.expr.ClassExpr;
 import com.github.javaparser.ast.expr.DoubleLiteralExpr;
@@ -221,6 +222,11 @@ public class FlowExpressionParseUtil {
             TypeMirror stringTM =
                     TypesUtils.typeFromClass(String.class, types, env.getElementUtils());
             return new ValueLiteral(stringTM, expr.asString());
+        }
+
+        @Override
+        public Receiver visit(BooleanLiteralExpr expr, FlowExpressionContext context) {
+            return new ValueLiteral(types.getPrimitiveType(TypeKind.BOOLEAN), expr.getValue());
         }
 
         /**

